### ISSUE TITLE
Prepare statement hangs when result set has no fields

### ISSUE THREAD

**Issue by Myrannas** [Philio/GoMySQL#38]
Running MySQL 5.1.49-1ubuntu8.1.

Upon create a prepared statement with the syntax below, the driver would hang. After enabling logging it appears the MySQL driver is waiting for another EOF packet to signify the end of the result fields.

> INSERT INTO User ( `Password` , `Username` ) VALUES (?,?)

Making a small modification to check if it should wait for another EOF packet as per below fixed the hang. This was just a very quick hack to restore functionality, I have not run further tests on it yet.

```
if stmt.ParamCount > 0 && !stmt.paramsEOF {
    stmt.paramsEOF = true
    if len(stmt.result.Fields)==0 {
        stmt.result.fieldsEOF = true
    }
    if mysql.Logging {
        log.Print("End of param packets")
    }
}
```

**Comment by Philio** [comment]
Hi there, which version are you using?

**Comment by Myrannas** [comment]
Hi Philio,

I am running version 0.2.7 from this repository.

**Comment by Philio** [comment]
Oh that's odd, that exact same scenario is in one of my test cases and I've got the same version on this machine...

$ mysql --version
mysql  Ver 14.14 Distrib 5.1.49, for debian-linux-gnu (x86_64) using readline 6.1

I'll run some more tests on this one, thanks for your report.

**Comment by Philio** [comment]
Confirmed and have added a fix, will push to the repo shortly and release.

**Comment by Philio** [comment]
A fix has been released, please update to 0.2.8
